### PR TITLE
Add `auto_device_count` method to `Accelerators`

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Follow up PR will be adding support for `devices="auto"`, where the accelerator will have the info what’s the maximum devices available to train on.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -688,3 +688,8 @@ def on_train_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int =
             "`on_train_batch_start` logic is implemented directly in the `TrainingTypePlugin` implementations."
         )
         return self.training_type_plugin.on_train_batch_start(batch, batch_idx)
+
+    @staticmethod
+    def get_devices_when_set_to_auto() -> int:
```
[the remark this answers is not in the document]
Comment (four4fish) [reply pytorch_lightning/accelerators/accelerator.py]: I was thinking about the same thing, should we rename devices to devices_count in accelerator_connector too?
https://github.com/PyTorchLightning/pytorch-lightning/issues/10231

[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/accelerators/accelerator.py]: @four4fish `devices` could be a list as well. For eg., [1] for TPUs means training on tpu core with id 1, whereas GPUs accepts list too.

[the remark this answers is not in the document]
Comment (four4fish) [reply pytorch_lightning/accelerators/accelerator.py]: > @four4fish `devices` could be a list as well. For eg., [1] for TPUs means training on tpu core with id 1, whereas GPUs accepts list too.

How about parallel_devices then?

**Review by awaelchli — APPROVED** [review]
